FuzzySearch fails on empty string?

Hey, kudos for the good work.

By the way, FuzzySearch fails on empty string input. I will look into your code and see if I can fix it.

```
package main

import "github.com/derekparker/trie"

func main() {
	t := trie.New()
	t.Add("foo", 1)
	t.FuzzySearch("")
}
```
Output -
```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/derekparker/trie.fuzzycollect(0xc000024080, 0xc000046688, 0x0, 0x20, 0x0, 0x20, 0x20)
	/home/amogh/go/src/github.com/derekparker/trie/trie.go:272 +0x4b0
github.com/derekparker/trie.Trie.FuzzySearch(0xc000024080, 0x1, 0x0, 0x0, 0x4aadf8, 0xc000024180, 0x0)
	/home/amogh/go/src/github.com/derekparker/trie/trie.go:117 +0x96
main.main()
	/home/amogh/go/src/tmp/tmp.go:8 +0xed
exit status 2
```

--- a/trie.go
+++ b/trie.go
@@ -265,6 +265,10 @@ type potentialSubtree struct {
 }
 
 func fuzzycollect(node *Node, partial []rune) []string {
+	if len(partial) == 0 {
+		return collect(node)
+	}
+
 	var (
 		m    uint64
 		i    int

--- a/trie_test.go
+++ b/trie_test.go
@@ -205,7 +205,6 @@ func TestPrefixSearch(t *testing.T) {
 }
 
 func TestFuzzySearch(t *testing.T) {
-	trie := New()
 	setup := []string{
 		"foosball",
 		"football",
@@ -230,18 +229,23 @@ func TestFuzzySearch(t *testing.T) {
 		{"fy", 1},
 		{"fz", 2},
 		{"a", 5},
+		{"", 8},
+		{"zzz", 0},
 	}
 
+	trie := New()
 	for _, key := range setup {
 		trie.Add(key, nil)
 	}
 
 	for _, test := range tests {
-		actual := trie.FuzzySearch(test.partial)
-		if len(actual) != test.length {
-			t.Errorf("Expected len(actual) to == %d, was %d for %s actual was %#v",
-				test.length, len(actual), test.partial, actual)
-		}
+		t.Run(test.partial, func(t *testing.T) {
+			actual := trie.FuzzySearch(test.partial)
+			if len(actual) != test.length {
+				t.Errorf("Expected len(actual) to == %d, was %d for %s actual was %#v",
+					test.length, len(actual), test.partial, actual)
+			}
+		})
 	}
 }
 

